flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/ReducingUpsertWriterTest.java

```diff
@@ -141,6 +142,73 @@ public static Object[] enableObjectReuse() {
                 TimestampData.fromInstant(Instant.parse("2021-03-30T21:00:00Z")))
     };
 
+    public static final RowData[] TEST_DATA_WITH_NULL_TIMESTAMP = {
+        GenericRowData.ofKind(
```

Comment:
I don't see why we need so many records. This should already fail even with a single record.
Reducing the number of required required records for the test would remove a lot of noise from the test.

Reply:
+1, I believe you can reuse `TEST_DATA` already defined in the class. `TEST_DATA` also has many records, but that's a different issue